### PR TITLE
docs: initial version of the documentation

### PR DESCRIPTION
**Pull request by mhevery**

### CODE REVIEW

File: modules/core/docs/view.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+ViewPort is location in a View which can accept child Views.
+
+* View is a core rendering construct. A running application is just a collection of Views which are nested in a tree like
+  structure. The View tree mimics the DOM tree.
+* Views represent a running instance of a DOM Template. This implies that while elements in a View can change
+  properties, they can not change structurally. (Structural changes such as, adding or removing elements requires
+  adding or removing child Views into ViewPorts.)
+* View can have zero or more ViewPorts. A ViewPort is a marker in the DOM which allows the insertion of child Views.
+* Views are created from a ProtoView. A ProtoView is a compiled DOM Template which is efficient at creating Views.
+* View contains a context object. The context represents the object instance against which all expressions are
```
Comment (vicb) [inline modules/core/docs/view.md]: against ... against

File: modules/core/docs/view.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+of the View. The instantiation process involves cloning the above template and locating all of the elements which
+contain bindings and finally instantiating the Directives associated with the template.
+(See compilation for more details.)
+
+```
+<div>                             | viewA
+  Your name:                      | viewA
+  <input var="name" type="Text">  | viewA: local variable 'name'
+  <br>                            | viewA
+  {{greeting}} {{name.value}}!    | viewA: binding expression 'name.value'
```
Comment (vicb) [inline modules/core/docs/view.md]: & `greeting`

File: modules/core/docs/view.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+</div>                            | viewA
+```
+
+The resulting View instance looks something like this (simplified pseudo code):
+```
+viewA = new View({
+  template: ...,
+  context: new Greeter(),
+  localVars: ['name'],
+  watchExp: ['name.value']
```
Comment (vicb) [inline modules/core/docs/view.md]: + `greeting`

File: modules/core/docs/view.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+class Greeter {
+  greeting:string;
+
+  constructor() {
+    this.greeting = 'Hello';
+  }
+}
+```
+
+And assume fallowing HTML Template:
```
Comment (vicb) [inline modules/core/docs/view.md]: type (following)

File: modules/core/docs/view.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+```
+<div>
+  Your name:
+  <input var="name" type="Text">
+  <br>
+  {{greeting}} {{name.value}}!
+</div>
+```
+
+The above template is compiled by the Compiler to create a ProtoView. The ProtoView is than used to create an instance
```
Comment (shidhincr) [inline modules/core/docs/view.md]: than or then ?

**Comment by mhevery** [comment]
Closed 829c28f

**Pull request closed without merging**